### PR TITLE
Support UUID for SQLAlchemy

### PR DESCRIPTION
**Pull request by lpoulain**
## Description

Provide a SQLAlchemy mapping to Trino's UUID type (resolves issue #331 )

## Non-technical explanation

Provide UUID support for SQLAlchemy

## Release notes

( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
(X) Release notes are required, with the following suggested text:

```markdown
* Support UUID for SQLAlchemy ({issue}`358`)
```

### CODE REVIEW

File: trino/sqlalchemy/datatype.py [hunk 1]
```diff
@@ -86,6 +86,11 @@ def get_col_spec(self, **kw):
         return 'JSON'
 
 
+class UUID(UserDefinedType):
```
Comment (hovaesco) [inline trino/sqlalchemy/datatype.py:89]: please add `__visit_name__`

**Review by hovaesco — APPROVED** [review]

File: trino/sqlalchemy/datatype.py (hunk 1 above)
Comment (lpoulain) [reply trino/sqlalchemy/datatype.py:89]: Done

File: tests/integration/test_sqlalchemy_integration.py [hunk 2]
```diff
@@ -89,13 +91,15 @@ def test_define_and_create_table(trino_connection):
         sqla.Table('users',
                    metadata,
                    sqla.Column('id', sqla.Integer),
+                   sqla.Column('guid', UUID),
```
Comment (mdesmet) [inline tests/integration/test_sqlalchemy_integration.py:94]: I think this should be sqla.Uuid instead. See https://docs.sqlalchemy.org/en/20/core/type_basics.html#sqlalchemy.types.Uuid

Comment (mdesmet) [reply tests/integration/test_sqlalchemy_integration.py]: Uuid only works in v2 of sqlalchemy. So you will need to ensure it only works in that version.

File: trino/sqlalchemy/datatype.py [hunk 3]
```diff
@@ -86,6 +86,13 @@ def get_col_spec(self, **kw):
...
+    __visit_name__ = "UUID"
+
+    def get_col_spec(self, **kw):
+        return "UUID"
+
+
```
Comment (mdesmet) [inline trino/sqlalchemy/datatype.py:95]: I think this code block should be removed as we don't need to add a custom `UserDefinedType`

File: trino/sqlalchemy/datatype.py [hunk 4]
```diff
@@ -122,7 +129,7 @@ def get_col_spec(self, **kw):
     #
     # === Mixed ===
     # 'ipaddress': IPADDRESS
-    # 'uuid': UUID,
+    "uuid": UUID,
```
Comment (mdesmet) [inline trino/sqlalchemy/datatype.py:132]: This should be mapped to the sqlalchemy uuid type if it's available

File: trino/sqlalchemy/compiler.py [hunk 5]
```diff
@@ -196,6 +196,9 @@ def visit_TIME(self, type_, **kw):
     def visit_JSON(self, type_, **kw):
         return 'JSON'
 
+    def visit_UUID(self, type, **kw):
+        return "UUID"
+
```
Comment (mdesmet) [inline trino/sqlalchemy/compiler.py:201]: This should only be added if the uuid type is available.

File: tests/unit/sqlalchemy/test_datatype_parse.py [hunk 6]
```diff
@@ -192,3 +192,8 @@ def test_parse_row(type_str: str, sql_type: ARRAY, assert_sqltype):
 def test_parse_datetime(type_str: str, sql_type: TypeEngine, assert_sqltype):
     actual_type = datatype.parse_sqltype(type_str)
     assert_sqltype(actual_type, sql_type)
+
+
+def test_parse_uuid():
+    actual_type = datatype.parse_sqltype("UUID")
+    assert type(actual_type) == UUID
```
Comment (mdesmet) [inline tests/unit/sqlalchemy/test_datatype_parse.py:199]: Test should only be succesful if type is available.

File: tests/integration/test_sqlalchemy_integration.py [hunk 7]
```diff
@@ -133,6 +135,60 @@ def test_insert(trino_connection):
         metadata.drop_all(engine)
 
 
+@pytest.mark.skipif(
+    sqlalchemy_version() < "2.0",
+    reason="columns argument to select() must be a Python list or other iterable"
```
Comment (mdesmet) [inline tests/integration/test_sqlalchemy_integration.py:140]: reason seems not correct.

**Review by mdesmet — COMMENTED** [review]
LGTM except reason

File: trino/sqlalchemy/datatype.py [hunk 8]
```diff
@@ -122,13 +123,16 @@ def get_col_spec(self, **kw):
     #
     # === Mixed ===
     # 'ipaddress': IPADDRESS
-    # 'uuid': UUID,
```
Comment (mdesmet) [inline trino/sqlalchemy/datatype.py:125]: Can you revert this change

**Comment by mdesmet** [comment]
Please also add that this resolves https://github.com/trinodb/trino-python-client/issues/331 in the PR description.

**Review by mdesmet — APPROVED** [review]

**Review by hashhar — APPROVED** [review]
